java/short_url_generator/Index.java

```diff
@@ -0,0 +1,280 @@
+ */
+private RuntimeResponse validatePayload(Map<String, Object> payload, RuntimeResponse res) {
+    Map<String, Object> responseData = new HashMap<>();
+
+    // Validate that payload has both provider and url
+    if(!payload.containsKey("provider") || !payload.containsKey("url")) {
+        responseData.put("success", false);
+        responseData.put("message", "Payload must contain both provider and url data");
+        return res.json(responseData);
+    }
+
+    String provider = payload.get("provider").toString();
+    String url = payload.get("url").toString();
+
+    // Validate the provider
+    if(!Provider.validateProvider(provider)) {
+        responseData.put("success", false);
+        String providerNames = Stream.of(Provider.values())
+                                    .map(Provider::getName)
+                                    .collect(Collectors.joining(", "));
+        responseData.put("message", "Provider " + provider + " is not supported currently. " +
+                                        "Only " + providerNames + " are supported");
+        return res.json(responseData);
+    }
+
+    // Validate the URL
+    UrlValidator urlValidator = new UrlValidator();
+    if (!urlValidator.isValid(url)) {
+        responseData.put("success", false);
+        responseData.put("message", "Provided URL: " + url + " is not valid, please provide a valid, correctly formed URL");
+        return res.json(responseData);
+    }
+    
+    return null;
+}
+
+/**
+ * This function will validate that the payload and variables are non-empty in the request
+ * 
+ * @param req is the received POST request
+ * @return null is nothing is empty, otherwise an error response
+ */
+private RuntimeResponse checkEmptyPayloadAndVariables(RuntimeRequest req, RuntimeResponse res) {
+    Map<String, Object> responseData = new HashMap<>();
+    
+    if(req.getPayload() == null || req.getPayload().trim().isEmpty() || req.getPayload().trim().equals("{}")) {
+        responseData.put("success", false);
+        responseData.put("message", "Payload is empty, expected a payload with provider and URL");
+        return res.json(responseData);
+    }
+
+    if(req.getVariables() == null) {
+        responseData.put("success", false);
+        responseData.put("message", "Empty function variables found. You need to pass an API key for the provider");
+        return res.json(responseData);
+    }
+    return null;
+}
+
+/**
+ * This method validates that a non-empty API key is present in variables
+ * 
+ * @param req is the received POST request
+ * @return null if non-empty API key is present, otherwise an error response
+ */
+private RuntimeResponse checkEmptyAPIKey(RuntimeRequest req, RuntimeResponse res) {
+    Map<String, String> variables = req.getVariables();
+
+    if(!variables.containsKey("apiKey")
+        || variables.get("apiKey") == null
+        || variables.get("apiKey").trim().isEmpty()) {
```

Comment:
This needs to change a little due to requested change in README. We should have 2 separate variables for each API key - for each provider. Then we only need to validate API key of a specific provider - that would either be `TINYURL_API_KEY` or `BITLY_API_KEY`.